setup.sh

```diff
@@ -116,17 +116,24 @@ sudo raspi-config nonint do_spi 0
 # been tested and verified.
 KERNEL_HEADERS_PACKAGE=raspberrypi-kernel-headers
 if ! dpkg -s $KERNEL_HEADERS_PACKAGE &> /dev/null; then
-    echo "Installing Raspberry Pi kernel headers"
-    sudo apt-get install -y $KERNEL_HEADERS_PACKAGE
+  echo "Installing Raspberry Pi kernel headers"
+  sudo apt-get install -y $KERNEL_HEADERS_PACKAGE
 fi
 
 echo "Installing the Python3 packages and related libs"
-sudo apt-get install -y python3-matplotlib
-sudo apt-get install -y python3-numpy
-sudo apt-get install -y libatlas-base-dev
+sudo apt-get install -y python3-matplotlib || ( echo "Error: Package installation failed" ; exit 1 )
```

Comment:
It would be useful for the error message to say which package failed to install.